core/src/main/scala/io/projectglow/sql/BigFileDatasource.scala

```diff
@@ -68,8 +68,10 @@ abstract class BigFileDatasource extends CreatableRelationProvider {
     }
 
     if (doSave) {
+      data.cache()
```

Comment:
We should uncache the RDD even if saving fails. Recommend introducing a wrapper that `withCachedRDD` / `withCachedDataset` to avoid repeating this logic.